spec/requests/search_spec.rb

```diff
@@ -96,22 +96,45 @@
   end
 
   describe "GET /search/usernames" do
-    let(:authorized_user) { create(:user) }
-    let(:result) do
-      {
-        "username" => "molly",
-        "name" => "Molly",
-        "profile_image_90" => "something"
-      }
+    before do
+      sign_in authorized_user
     end
 
-    it "returns json" do
-      sign_in authorized_user
-      allow(Search::User).to receive(:search_usernames).and_return(
-        result,
-      )
-      get "/search/usernames"
-      expect(response.parsed_body).to eq("result" => result)
+    context "when using Elasticsearch" do
+      let(:result) do
+        {
+          "username" => "molly",
+          "name" => "Molly",
+          "profile_image_90" => "something"
+        }
+      end
+
+      it "returns json" do
+        allow(Search::User).to receive(:search_usernames).and_return(
+          result,
+        )
+        get "/search/usernames"
+        expect(response.parsed_body).to eq("result" => result)
+      end
+    end
+
+    context "when using PostgreSQL" do
+      before do
+        allow(FeatureFlag).to receive(:enabled?).with(:search_2_usernames).and_return(true)
+      end
+
+      it "returns nothing if there is no username parameter" do
+        get search_usernames_path
+        expect(response.parsed_body["result"]).to be_empty
+      end
```

Comment:
We could, yes! That's a valid point.

I was trying to remain consistent with the pattern we have going:

https://github.com/forem/forem/blob/f1d85e666bb651fdeb3e91a4c8d8c5085af7b1cd/spec/requests/search_spec.rb#L41-L44

I'd rather be consistent across searches and I'm hesitant to keep introducing more changes on each PR because then at the end they're all different.

Maybe this is something we revisit at the end in a larger refactor? That's how we did it when implemented Elasticsearch originally.

cc: @rhymes